apps/services/search-indexer/src/migrate/lib/aws.ts

```diff
@@ -43,66 +69,64 @@ interface PackageStatuses {
 const getPackageStatuses = async (
   packageId: string,
 ): Promise<PackageStatuses> => {
-  const params = {
+  const describePackagesCommand = new DescribePackagesCommand({
     Filters: [
       {
         Name: 'PackageID',
         Value: [packageId],
       },
     ],
-  }
-  const packages = await awsEs.describePackages(params).promise()
+  })
+  const packages = await awsEs.send(describePackagesCommand)
 
-  const domainPackageList = await awsEs
-    .listPackagesForDomain({
-      DomainName: environment.esDomain,
-    })
-    .promise()
+  const listPackagesForDomainCommand = new ListPackagesForDomainCommand({
+    DomainName: environment.esDomain,
+  })
+  const domainPackageList = await awsEs.send(listPackagesForDomainCommand)
 
-  const domainPackage = domainPackageList.DomainPackageDetailsList.find(
+  const domainPackage = domainPackageList.DomainPackageDetailsList?.find(
     (listItem) => listItem.PackageID === packageId,
   )
 
   return {
-    packageStatus: packages.PackageDetailsList[0].PackageStatus,
-    domainStatus: domainPackage?.DomainPackageStatus ?? 'DISSOCIATED',
+    packageStatus: packages.PackageDetailsList?.[0]
+      .PackageStatus as PackageStatus,
+    domainStatus: domainPackage?.DomainPackageStatus ?? 'DISSOCIATING',
```

Comment:
I assume the `DISSOCIATED` → `DISSOCIATING` was caught because of the es client switchout/update?
So I'm further assuming that the type system will bark at us if `AVAILABLE` and `DISSOCIATING` are misspelt?

If not, I'd like to see these enumerated somewhere :>

Reply:
Well spotted, these types are already defined in PackageStatus and DomainPackageStatus so I modified the code to use those.